### PR TITLE
Toilet BUFF

### PR DESCRIPTION
**Pull request by lbnesquik**
Make it possible to put reagents into a toilet cistern, allowing for tained swirlies. Wanna do acid swirlies?
You sure can! Also there is now a flushing noise. Alt click to flush.

### CODE REVIEW

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -47,12 +51,29 @@
...
 
+	if(I.is_wrench())
+		water_supply = !water_supply
+		to_chat(user, "You [water_supply ? "close" : "open"] \the [src] water supply. It won't refill after being flushed now.")
+		return //So you don't put the wrench insides
+
+	if(istype(I, /obj/item/weapon/reagent_containers))
+		if(cistern)
+			if(I.reagents.total_volume < 1)
+				user << "\The [I] is empty."
```
Comment (Poojawa) [inline code/game/objects/structures/watercloset.dm]: to_chat(user, instead of <<

File: code/game/objects/structures/watercloset.dm [hunk 2]
```diff
@@ -47,12 +51,29 @@
...
+				return
+			else
+				to_chat(user, "<span class='notice'>You empty 20 units out of \the [I.name] into \the [src].</span>")
+				I.reagents.trans_to_obj(src, 20)
+				return
+		else
+			user << "\The [src] cistern is closed."
```
Comment (Poojawa) [inline code/game/objects/structures/watercloset.dm]: to_chat(user, instead of <<

File: code/game/objects/structures/watercloset.dm (hunk 1 above)
Comment (lbnesquik) [reply code/game/objects/structures/watercloset.dm]: what are you doing reviewing pr's 'round here, partner

File: code/game/objects/structures/watercloset.dm [hunk 3]
```diff
@@ -47,12 +51,29 @@
...
-		if(do_after(user, 30))
+		if(do_after(user, 15))
 			user.visible_message("<span class='notice'>[user] [cistern ? "replaces the lid on the cistern" : "lifts the lid off the cistern"]!</span>", "<span class='notice'>You [cistern ? "replace the lid on the cistern" : "lift the lid off the cistern"]!</span>", "You hear grinding porcelain.")
 			cistern = !cistern
 			update_icon()
 			return
 
+	if(I.is_wrench())
+		water_supply = !water_supply
+		to_chat(user, "You [water_supply ? "close" : "open"] \the [src] water supply. It won't refill after being flushed now.")
```
Comment (Screemonster) [inline code/game/objects/structures/watercloset.dm]: "You open the water supply. It won't refill after being flushed now."

????

File: code/game/objects/structures/watercloset.dm [hunk 4]
```diff
@@ -47,12 +51,29 @@
...
+		return //So you don't put the wrench insides
+
+	if(istype(I, /obj/item/weapon/reagent_containers))
+		if(cistern)
+			if(I.reagents.total_volume < 1)
+				to_chat(user, "\The [I] is empty.")
+				return
+			else
+				to_chat(user, "<span class='notice'>You empty 20 units out of \the [I.name] into \the [src].</span>")
```
Comment (Screemonster) [inline code/game/objects/structures/watercloset.dm]: Why not use the transfer volume from whatever the container is?

Comment (Arokha) [reply code/game/objects/structures/watercloset.dm]: Which is the extremely long `var/amount_per_transfer_from_this`

**Comment by SpadesNeil** [comment]
I sit here and think, "This is the kind of nonsense Vorestation comes up with."

**Comment by Screemonster** [comment]
to be fair I've thought giving someone a polytrinic swirly on /tg/ would be a brutal way for a traitor to kill their target. Hell, I'd probably trade my telecrystals for an unanchored toilet if I rolled traitor just for the spectacle of dragging a toilet around the halls and swirlying people to death in it.

**Comment by lbnesquik** [comment]
@mistyLuminescence This doesn't interact at all. You would empty the beaker for sure.

File: code/game/objects/structures/watercloset.dm (hunk 3 above)
Comment (lbnesquik) [reply code/game/objects/structures/watercloset.dm]: Electrical terms. Circuit being open = not complete = not working = not going to refill. Im going to change that.

File: code/game/objects/structures/watercloset.dm (hunk 4 above)
Comment (lbnesquik) [reply code/game/objects/structures/watercloset.dm]: I am not sure how to do that.

**Comment by Atermonera** [comment]
So it DOES interact by outright removing that function for reagent containers?

**Comment by lbnesquik** [comment]
Er, *yes?* You can't hide beakers into toilets, or any reagent containers for that matter. I felt like adding another variable to control whether you pour the contents or put the beaker in would had been too complex and frankly, unneeded.

**Comment by Screemonster** [comment]
If it's already possible to hide beakers then continue to make them hideable _only_ if the lid is on.

Seriously, you're putting an open container into a toilet cistern. Full of water. The contents of that container are not gonna stay contained.

**Comment by lbnesquik** [comment]
But @Screemonster  im lazyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy. 
I doubt anyone ever hid any beaker of liquid anyways. How often do people get reagents that they want to hide in toilets? The only reason you do that is if you are expecting to be arrested. But, if there is insistent request, i'll make it so you can hide beakers as well.

File: code/game/objects/structures/watercloset.dm [hunk 5]
```diff
@@ -47,12 +51,30 @@
...
-		if(do_after(user, 30))
+		if(do_after(user, 15))
 			user.visible_message("<span class='notice'>[user] [cistern ? "replaces the lid on the cistern" : "lifts the lid off the cistern"]!</span>", "<span class='notice'>You [cistern ? "replace the lid on the cistern" : "lift the lid off the cistern"]!</span>", "You hear grinding porcelain.")
 			cistern = !cistern
 			update_icon()
 			return
 
+	if(I.is_wrench())
+		water_supply = !water_supply
+		to_chat(user, "You [water_supply ? "open" : "close"] \the [src] water supply valve. It won't refill after being flushed now.")
```
Comment (Screemonster) [inline code/game/objects/structures/watercloset.dm]: Still says "it won't refill after being flushed now" regardless of the open/closed state.
"you open the valve, now it won't refill"
"you close the valve, now it won't refill"

Comment (lbnesquik) [reply code/game/objects/structures/watercloset.dm]: Oh i get the issue now.

**Comment by lbnesquik** [comment]
Now it works btw.

File: code/game/objects/structures/watercloset.dm [hunk 6]
```diff
@@ -47,12 +51,33 @@
...
+	if(I.is_wrench())
+		water_supply = !water_supply
+		if(water_supply)
+			to_chat(user, "You open \the water supply valve. It can flush again now.")
+		else
+			to_chat(user, "You close \the [src] water supply valve. It won't refill after being flushed now.")
+		return //So you don't put the wrench insides
+
+	if(istype(I, /obj/item/weapon/reagent_containers))
+		if(cistern)//If 1, it is open.
```
Comment (Screemonster) [inline code/game/objects/structures/watercloset.dm]: Wait, so you have to crowbar the cistern open to add reagents? Not just pour them into the bowl?

**Pull request closed without merging**